certbot/certbot/_internal/renewal.py

```diff
@@ -390,8 +390,9 @@ def renew_cert(config: configuration.NamespaceConfig, domains: Optional[List[str
     else:
         prior_version = lineage.latest_common_version()
         # TODO: Check return value of save_successor
-        lineage.save_successor(prior_version, new_cert, new_key.pem, new_chain, config)
+        version = lineage.save_successor(prior_version, new_cert, new_key.pem, new_chain, config)
         lineage.update_all_links_to(lineage.latest_common_version())
+        lineage.truncate(version, prior_version)
```

Comment:
In principle, I think that `truncate` should not make Certbot crash, because it is called at the completion of `certbot renew`. I feel compelled to put a broad `except` here, or maybe to catch and ignore errors aggressively within `truncate` itself.

There are a lot of errors (I/O ones) that can be raised though. Not sure what the best strategy is here.

Reply:
Other than `OSError`, the other way this can crash is if something is wrong with the lineage symlinks or similar. However, since we literally just succeeded in renewing the certificate and updating the links, that possibility should hopefully be excluded.

So I am personally comfortable with just the `OSError` approach.